infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -320,13 +320,9 @@ private void handleTokenQuery(CorfuPayloadMsg<TokenRequest> msg,
             // Global tail query
             token = new Token(sequencerEpoch, globalLogTail - 1);
             streamTails = Collections.emptyList();
-        } else if (req.getStreams().size() == 1) {
-            // single stream query
-            token = new Token(sequencerEpoch, streamTailToGlobalTailMap.getOrDefault(streams.get(0), Address.NON_EXIST));
-            streamTails = Collections.emptyList();
         } else {
-            // multiple stream query, the token is populated with the global tail and the tail queries are stored in
-            // streamTails
+            // multiple or single stream query, the token is populated with the global tail
+            // and the tail queries are stored in streamTails
             token = new Token(sequencerEpoch, globalLogTail - 1);
```

Comment:
I think a map is a more appropriate data structure. That way the consumer doesn't have to maintain a uuid -> index

Reply:
Yeah, this is old code. I just took what was there but this is more convenient. Will do.

Reply:
Done.